core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -88,6 +93,18 @@ public TableScan appendsAfter(long fromSnapshotId) {
           String.format("Cannot incrementally scan table of type %s", type.name()));
     }
 
+    @Override
+    public CloseableIterable<FileScanTask> planFiles() {
+      if (type.equals(MetadataTableType.ALL_DATA_FILES)) {
```

Comment:
I had to move this code from BaseAllMetadataTableScan.  

I didn't realize that there was an alternate base class for "All*" metadata tables with special handling (it seems for WAP case).  As there's no multiple inheritance for Java, I can't inherit from both BaseFileTable and BaseAllMetadataTable scans, so I decided to copy the logic.

An alternative is to use composition, let me know if that's a better way, as I thought it's a bit overkill for a few lines.  As a consequence now though, the BaseAllMetadataTableScan supports only 'all_entries' table.

Reply:
Refactored to make it slightly cleaner.